rotkehlchen/chain/evm/contracts.py

```diff
@@ -131,6 +136,105 @@ def decode_event(
         return decode_event_data_abi(tx_log=tx_log, event_abi=event_abi)  # type: ignore
 
 
+def try_download_abi_from_remote(name: str) -> Optional[list[dict[str, Any]]]:
+    """
+    Checks whether abi with the given name exists on remote.
+    If it does, downloads it and adds to globaldb.
+
+    Returns the abi if found and None otherwise.
+    """
+    abi_url = f'{BASE_CONTRACT_DATA_URL}/abi/{name}.json'
+    try:
+        abi_data = request_get_list(abi_url)
+    except (RemoteError, UnableToDecryptRemoteData) as e:
+        log.error(f'Could not find abi {name} in rotki data repo due to: {str(e)}')
+        return None
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        write_cursor.execute(
+            'INSERT INTO contract_abi(name, value) VALUES(?, ?)',
+            (name, json.dumps(abi_data, separators=(',', ':'))),
+        )
+
+    return abi_data
+
+
+class ContractSchema(Schema):
+    address = fields.String(required=True)  # not using EvmAddressField since it would cause circular imports  # noqa: E501
```

Comment:
you can move this schema to `serialization/schemas.py` and should be fine I guess